### ISSUE TITLE
[Feature Request] Inline & Multiline Comments

### ISSUE THREAD

**Issue by chipsenkbeil** [vimwiki/vimwiki#965]
Another feature request here, not a bug. Making a note of this myself in case I'm able to dive in to offer this functionality. I'm also brand new to vimwiki, so there may be existing functionality that can do this or maybe people have existing workflows to achieve a similar result.

### Request

Support comments that are inline and/or multiline versus being at the start of a line.

```vimwiki
= My Wiki =

%% Line comment
- Item 1
- Item 2 %%+ Inline comment +%%
- Item 3
%%+ some comment
that continues
on multiple lines +%%
```

### Reason

I'm looking around at ways that other text-based languages handle extra metadata. Org mode has a variety of syntaxes like properties, tags, etc. that extensions like org-pomodoro leverage to mark tasks with metadata. The closest I've seen in vimwiki are tags, which also act as anchors; however, stashing extra metadata doesn't seem as approachable.

I'm not saying supporting metadata or any advanced functionality needs to be baked into vimwiki itself. Rather, providing inline comments means that plugins that want to extend vimwiki could stash extra info in the same line as a task, heading, etc. beyond the labelling that a tag provides. This also avoids tacking on information to the actual content when it's rendered into HTML or other formats where you don't want it to be included.

### Example Use Case

If I was to implement a plugin that extends vimwiki to support the pomodoro technique, one way would be to do the following:

1. Detect if cursor is on a task
2. Prepend some tags to indicate being used with pomodoro for searching/identification
3. Append an inline comment that contents metadata associated with the task, such as the start time, time spent, or pomodoro achieved

```vimwiki
- [ ] :pomodoro:active: Some task that is actively being tracked %%+ {"time": "10 minutes"} +%%
- [ ] :pomodoro: Some task that has been tracked, but is not active and not completed %%+ {"time": "7 minutes"} +%%
- [X] :pomodoro:complete: Some task that has been completed %%+ {"time": "38 minutes", "pomodoro_count": "1" } +%% 
- [ ] Some task not being tracked for time
```

Of course, the plugin would collapse the metadata to hide it (similar to how links are collapsed until you move over the line.

```vimwiki
- [ ] :pomodoro:active: Some task that is actively being tracked %%+ ... +%%
- [ ] :pomodoro: Some task that has been tracked, but is not active and not completed %%+ ... +%%
- [X] :pomodoro:complete: Some task that has been completed %%+ ... +%% 
- [ ] Some task not being tracked for time
```

This can be done today with pure tags (if you don't need specific details), line comments above/below and some fancy wizardry, or using a separate database in combination with tags to keep track of which task matches which subset of data. Seems like being able to try out new features by encoding information in inline comments could open up some interesting doors.

### Requirements

Minimum for my use case would be the ability to add line comments later in a line.

```vimwiki
- Some text %% comment rest of line
```

Full-blown solution would be able to use comments with a start and stop breakdown.

```vimwiki
- Some text %%+ inline comment +%% more text
```

**Comment by chipsenkbeil** [comment]
## Progress Tracker

### Vim Syntax Highlighting

Started with something simple, which is updating the syntax highlighting for vim to support an inline comment. Going with the `%%+ something +%%` style for now. Editing `vimwiki/syntax/vimwiki_default.vim`:

```viml
" Matches %%+ <ZERO OR MORE> +%% across multiple lines with closing +%% being                                                                     
" optional (as you can have a dangling comment) or keeps the line comment                                                                         
" starting with %% as long as it doesn't have a + immediately after                                                                               
let s:default_syntax.rxComment = '%%+\%(\_.\{-\}+%%\|\_.*\)\|^\s*%%\%([^+].*\)\?$'
```

**Comment by tinmarino** [comment]
Very nice request. Not the feature itself but the way to integrate with your plugin permanently.

Maybe we should add a list of optional plugin and an easyer way for the user to integrate them and for vimwiki to not break support for those.

Your help would be really appreciated for this pionner interface

**Comment by chipsenkbeil** [comment]
@tinmarino, once we get this landed, I plan to try to build out the pomodoro functionality mentioned above as a plugin that you slide in alongside vimwiki. I'm sure I'll hit pain points as I go through that process and will be happy to contribute to ironing out a proper API, methodology, etc. for extending vimwiki in the same way that folks extend orgmode.

Just let me know in what ways you'd like me to contribute :)

**Comment by tinmarino** [comment]
> Just let me know in what ways you'd like me to contribute :)

Starting an issue when we need to speak about that, and explaining clearly what you need, from an high level point of view, so we get clear before implementing

**Comment by tinmarino** [comment]
Fixed : #967 your PR.
Nice job!
